e2e/whisper/whisper_mailbox_test.go

```diff
@@ -231,10 +230,395 @@ func (s *WhisperMailboxSuite) TestRequestMessageFromMailboxAsync() {
 
 }
 
-func (s *WhisperMailboxSuite) startBackend() (*api.StatusBackend, func()) {
-	//Start sender node
+func (s *WhisperMailboxSuite) TestRequestMessagesFromMailboxFromGroupChat() {
+	//Start mailbox, alice, bob, charlie node
+	mailboxBackend, stop := s.startMailboxBackend()
+	defer stop()
+	mailboxNode, err := mailboxBackend.NodeManager().Node()
+	s.Require().NoError(err)
+	mailboxEnode := mailboxNode.Server().NodeInfo().Enode
+
+	aliceBackend, stop := s.startBackend("alice")
+	defer stop()
+	aliceNode, err := aliceBackend.NodeManager().Node()
+	s.Require().NoError(err)
+	aliceEnode := aliceNode.Server().NodeInfo().Enode
+
+	bobBackend, stop := s.startBackend("bob")
+	defer stop()
+
+	charlieBackend, stop := s.startBackend("charlie")
+	defer stop()
+
+	s.Require().NotEqual(mailboxEnode, aliceEnode)
+
+	err = aliceBackend.NodeManager().AddPeer(mailboxEnode)
+	s.Require().NoError(err)
+	err = bobBackend.NodeManager().AddPeer(mailboxEnode)
+	s.Require().NoError(err)
+	err = charlieBackend.NodeManager().AddPeer(mailboxEnode)
+	s.Require().NoError(err)
+	//wait async processes on adding peer
+	time.Sleep(time.Second)
+
+	aliceWhisperService, err := aliceBackend.NodeManager().WhisperService()
+	s.Require().NoError(err)
+	bobWhisperService, err := bobBackend.NodeManager().WhisperService()
+	s.Require().NoError(err)
+	charlieWhisperService, err := charlieBackend.NodeManager().WhisperService()
+	s.Require().NoError(err)
+
+	//add mailserver key
+	password := "status-offline-inbox"
+	bobMailServerKeyID, err := bobWhisperService.AddSymKeyFromPassword(password)
+	s.Require().NoError(err)
+	charlieMailServerKeyID, err := charlieWhisperService.AddSymKeyFromPassword(password)
+	s.Require().NoError(err)
+
+	//get rpc client
+	aliceRpcClient := aliceBackend.NodeManager().RPCClient()
+	s.Require().NotNil(aliceRpcClient)
+	bobRpcClient := bobBackend.NodeManager().RPCClient()
+	s.Require().NotNil(bobRpcClient)
+	charlieRpcClient := charlieBackend.NodeManager().RPCClient()
+	s.Require().NotNil(charlieRpcClient)
+
+	//generate group chat symkey and topic
+	groupChatKeyID, err := aliceWhisperService.GenerateSymKey()
+	s.Require().NoError(err)
+	groupChatKey, err := aliceWhisperService.GetSymKey(groupChatKeyID)
+	s.Require().NoError(err)
+	//generate group chat topic
+	groupChatTopic := whisperv5.BytesToTopic([]byte("groupChatTopic"))
+	groupChatPayload := newGroupChatParams(groupChatKey, groupChatTopic)
+	payloadStr, err := groupChatPayload.Encode()
+	s.Require().NoError(err)
+
+	//Add bob and charlie create key pairs to receive symmetric key for group chat from alice
```

Comment:
I think that for simplicity you can assume that everyone has a symmetric key they will use for the group chat.

Reply:
Yes, it's example of symmetric key transfer. Alice create symmetric key and send it to bob and charlie using asymmetic encryption(bob and charlie pubkeys)

Reply:
What I mean is that a symmetric key is known to all parties beforehand so there is no need to exchanging it at all. It may simplify this scenario.

Reply:
Pinging about this one. Still, I don't see why we need to exchange keys instead of assuming each party already has a symmetric key used in group chat.

Reply:
@adambabik I don't get you point. Could you explain a bit? For example, Alice want to start group chat with Bob and Charlie. How Bob and Charlie should get symmetric key and topic for group chat?

Reply:
>  Did you talk about skiping symkey exchange in this test?

Exactly. It's not relevant to this test as it's all about requesting cached messages from the MailServer. We can safely assume that the symmetric key is already exchanged and known to all parties, right?

Reply:
@adambabik Yes, we can. But one of idea of this test is to keep how participants of group chat receive symkey. I think it could be helpful it because receiving symkey looks not obvious for me.

Reply:
It's not obvious but it's also not a status-go job. Exchanging a symmetric key may be realized in a totally different way by status-react and that's fine. 

We can leave it if you insist but think about the confusion it can introduce as this step is not relevant to MailServer at all.